core/src/main/java/org/bitcoinj/core/Message.java

```diff
@@ -26,18 +30,33 @@ public interface Message {
      */
     int MAX_SIZE = 0x02000000;
 
+    /**
+     * Serializes this message to the provided stream. If you just want the raw bytes use {@link #serialize()}.
+     */
+    void bitcoinSerializeToStream(OutputStream stream) throws IOException;
```

Comment:
This exposing Java I/O to consumers is what I've been trying to avoid. I'd propose migrating to a ByteBuffer-based API first (step 2 from https://github.com/bitcoinj/bitcoinj/pull/3659#issuecomment-2754728681).

How about `ByteBuffer write(ByteBuffer buf) throws BufferOverflowException`, in line with many of our non-message data classes?

Reply:
Agreed. We'll do that refactoring first.